pkg/trafficmngr/nftables/nftables.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2024 flannel authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//go:build !windows
+// +build !windows
+
+package nftables
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	log "k8s.io/klog/v2"
+
+	"github.com/flannel-io/flannel/pkg/ip"
+	"github.com/flannel-io/flannel/pkg/lease"
+	"sigs.k8s.io/knftables"
+)
+
+const (
+	ipv4Table    = "flannel-ipv4"
+	ipv6Table    = "flannel-ipv6"
+	forwardChain = "forward"
+	postrtgChain = "postrtg"
+	//maximum delay in second to clean-up when the context is cancelled
+	cleanUpDeadline = 15
+)
+
+type NFTablesManager struct {
+	nftv4 knftables.Interface
+	nftv6 knftables.Interface
+}
+
+func (nftm *NFTablesManager) Init(ctx context.Context, wg *sync.WaitGroup) error {
+	var err error
+	nftm.nftv4, err = initTable(ctx, knftables.IPv4Family, ipv4Table)
+	if err != nil {
+		return err
+	}
+	nftm.nftv6, err = initTable(ctx, knftables.IPv6Family, ipv6Table)
+	if err != nil {
+		return err
+	}
+
+	wg.Add(1)
+	go func() {
+		<-ctx.Done()
+		log.Info("Cleaning-up flannel tables...")
+
+		cleanupCtx, cleanUpCancelFunc := context.WithTimeout(context.Background(), cleanUpDeadline*time.Second)
+		defer cleanUpCancelFunc()
+		err := nftm.cleanUp(cleanupCtx)
+		log.Errorf("nftables: error while cleaning-up: %v", err)
+		wg.Done()
+	}()
+	return nil
+}
+
+func initTable(ctx context.Context, ipFamily knftables.Family, name string) (knftables.Interface, error) {
```

Comment:
could you add a one-liner description for the function following [go best practices](https://tip.golang.org/doc/comment#:~:text=A%20function's%20doc%20comment%20should,any%20special%20syntax%20like%20backquotes.)

Reply:
done